be/src/storage/lake/spill_mem_table_sink.cpp

```diff
@@ -139,11 +147,102 @@ Status SpillMemTableSink::flush_chunk_with_deletes(const Chunk& upserts, const C
     return Status::OK();
 }
 
+class BlockGroupIterator : public ChunkIterator {
+public:
+    BlockGroupIterator(SpillMemTableSink& sink, Schema schema, const std::vector<spill::BlockPtr>& blocks)
+            : ChunkIterator(schema), _sink(sink), _blocks(blocks) {}
+
+    virtual Status do_get_next(Chunk* chunk) override {
+        if (_block_idx >= _blocks.size()) {
+            return Status::EndOfFile("End of block group");
+        }
+        RETURN_IF_ERROR(_sink.read_chunk_from_group_block(chunk, *_blocks[_block_idx]));
+        _block_idx++;
+        return Status::OK();
+    }
+
+    void close() {}
+
+private:
+    SpillMemTableSink& _sink;
+    const std::vector<spill::BlockPtr>& _blocks;
+    size_t _block_idx = 0;
+};
+
+Status SpillMemTableSink::read_chunk_from_group_block(Chunk* chunk, spill::Block& block) {
+    spill::SerdeContext ctx;
+    spill::BlockReaderOptions options;
+    auto& metrics = _spiller->serde()->parent()->metrics();
+    options.read_io_timer = metrics.read_io_timer;
+    options.read_io_count = metrics.read_io_count;
+    options.read_io_bytes = metrics.restore_bytes;
+    auto reader = block.get_reader(options);
+    DCHECK(reader);
+    ASSIGN_OR_RETURN(auto tmp_chunk, _spiller->serde()->deserialize(ctx, reader.get()));
+    tmp_chunk->swap_chunk(*chunk);
+    return Status::OK();
+}
+
 Status SpillMemTableSink::merge_blocks_to_segments() {
```

Comment:
I think it's better to use a threadpool for merge jobs, otherwise delta writer bthread will be hung and affect ingestion task which doesn't need to do merge blocks.

Reply:
will optimize in future pr